### DISCUSSION TITLE
Recording system audio

### DISCUSSION THREAD

**Discussion by jcutrell** [8ta4/say#163] [Ideas]
Hi folks!

I wear headphones virtually all the time, I'd like to record audio not only from my input audio, but also from my output audio. Is this something that is supported or even possible? I assume there would need to be some kind of combined sum source audio to set as the source for `say`, but I'm not sure.

**Comment by 8ta4** [comment]
I also wear headphones virtually, all the time. Everyone's in one these days! It's called an echo chamber.

If you want to record both your mic and your system audio on macOS, here's the quick rundown:

1. Install [BlackHole](https://github.com/ExistentialAudio/BlackHole). This virtual audio driver lets you route system audio into `say`.

2. [Create an Aggregate Device](https://support.apple.com/en-us/102171). This combines your physical mic and the virtual audio feed from BlackHole.

3. [Create a Multi-Output Device](https://support.apple.com/guide/audio-midi-setup/play-audio-through-multiple-devices-at-once-ams7c093f372/mac#:~:text=In%20the%20Audio%20MIDI%20Setup,then%20enter%20a%20new%20name.). You can still hear everything in your headphones while sending audio to `say`.

4. Select the Aggregate Device in `say`.

5. Make sure you're not in a hideaway.
